krun/platform.py

```diff
@@ -914,15 +917,17 @@ def check_preliminaries(self):
 
         if self.config.ENABLE_PINNING:
             self._check_cset_installed()
+        # Clear any leftover cpu pinning since it will break taskset
+        self.clear_cpu_pinning()
         self._check_isolcpus()
-        self._check_cset_shield()
         self._check_cpu_governor()
         self._check_cpu_scaler()
         self._check_perf_samplerate()
         if not self.no_tickless_check:
             self._check_tickless_kernel()
         self._check_aslr_enabled()
         self._check_dmesg_unrestricted()
+        self._check_cset_shield()
```

Comment:
And this will need to be guarded, as it doesn't check for `ENABLE_PINNING` itself. When `ENABLE_PINNING` is false there is no guarantee the cset userspace tool is installed.